src/node_wasi.cc

```diff
@@ -74,22 +74,55 @@ using v8::ArrayBuffer;
 using v8::BackingStore;
 using v8::BigInt;
 using v8::Context;
+using v8::Exception;
 using v8::FunctionCallbackInfo;
 using v8::FunctionTemplate;
+using v8::Integer;
+using v8::Isolate;
 using v8::Local;
 using v8::Object;
 using v8::String;
 using v8::Uint32;
 using v8::Value;
 
 
+static Local<Value> WASIException(Isolate* isolate,
+                                  int errorno,
+                                  const char* syscall) {
+  Environment* env = Environment::GetCurrent(isolate);
```

Comment:
Can we pass the `Context` explicitly instead?

Reply:
There is couple of places use `isolate`, and we can get `Context` from isolate. Any pitfalls I am not aware of ?

Reply:
Yes – partially, it’s a good practice to always pass the object with the most specific information here (i.e. prefer `Environment` over `Isolate`, `Context` over `Environment`). Also, more practically speaking, it would be good to get rid of the assumption that `env->context()` and `isolate->GetCurrentContext()` can be used interchangeably, e.g. for implementing something like https://github.com/nodejs/node/issues/28823.